observability/src/test/java/io/grpc/observability/interceptors/GcpLogHelperTest.java

```diff
@@ -0,0 +1,565 @@
+    String methodName = "method";
+    String authority = "authority";
+    Duration timeout = Durations.fromMillis(1234);
+    String rpcId = "d155e885-9587-4e77-81f7-3aa5a443d47f";
+    InetAddress address = InetAddress.getByName("127.0.0.1");
+    int port = 12345;
+    InetSocketAddress peerAddress = new InetSocketAddress(address, port);
+
+    GrpcLogRecord.Builder builder =
+        metadataToProtoTestHelper(EventType.GRPC_CALL_REQUEST_HEADER, nonEmptyMetadata)
+            .toBuilder()
+            .setTimestamp(timestamp)
+            .setSequenceId(seqId)
+            .setServiceName(serviceName)
+            .setMethodName(methodName)
+            .setEventType(EventType.GRPC_CALL_REQUEST_HEADER)
+            .setEventLogger(EventLogger.LOGGER_CLIENT)
+            .setLogLevel(LogLevel.LOG_LEVEL_DEBUG)
+            .setRpcId(rpcId);
+    builder.setAuthority(authority)
+        .setTimeout(timeout);
+    GrpcLogRecord base = builder.build();
+
+    // logged on client
+    {
+      sinkWriter.logClientHeader(
+          seqId,
+          serviceName,
+          methodName,
+          authority,
+          timeout,
+          nonEmptyMetadata,
+          EventLogger.LOGGER_CLIENT,
+          rpcId,
+          null);
+      verify(sink).write(base);
+    }
+
+    // logged on server
+    {
+      sinkWriter.logClientHeader(
+          seqId,
+          serviceName,
+          methodName,
+          authority,
+          timeout,
+          nonEmptyMetadata,
+          EventLogger.LOGGER_SERVER,
+          rpcId,
+          peerAddress);
+      verify(sink).write(
+          base.toBuilder()
+              .setPeerAddress(GcpLogHelper.socketToProto(peerAddress))
+              .setEventLogger(EventLogger.LOGGER_SERVER)
+              .build());
+    }
+
+    //authority is null
+    {
+      sinkWriter.logClientHeader(
+          seqId,
+          serviceName,
+          methodName,
+          null,
+          timeout,
+          nonEmptyMetadata,
+          EventLogger.LOGGER_CLIENT,
+          rpcId,
+          null);
+      verify(sink).write(
+          base.toBuilder()
+              .clearAuthority()
+              .build());
+    }
+
+    // timeout is null
+    {
+      sinkWriter.logClientHeader(
+          seqId,
+          serviceName,
+          methodName,
+          authority,
+          null,
+          nonEmptyMetadata,
+          EventLogger.LOGGER_CLIENT,
+          rpcId,
+          null);
+      verify(sink).write(
+          base.toBuilder()
+              .clearTimeout()
+              .build());
+    }
+
+    // peerAddress is not null (error on client)
+    try {
+      sinkWriter.logClientHeader(
+          seqId,
+          serviceName,
+          methodName,
+          authority,
+          timeout,
+          nonEmptyMetadata,
+          EventLogger.LOGGER_CLIENT,
+          rpcId,
+          peerAddress);
+      fail();
+    } catch (IllegalArgumentException expected) {
+      // noop
```

Comment:
You should check the contents of `expected` such as message etc

Reply:
Updated